runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -215,7 +215,12 @@ impl<Accounts: modules::accounts::API, Consensus: modules::consensus::API>
 
         // Signer.
         let signer = &ctx.tx_auth_info().signer_info[0];
-        Consensus::ensure_compatible_tx_signer(ctx)?;
+        if body.to.is_none() {
+            // If no `to` field is specified, i.e. withdrawing to the transaction sender's account,
+            // only allow the consensus-compatible single-Ed25519-key signer type. Otherwise, the
+            // tokens would get stuck in an account that you can't sign for on the consensus layer.
+            Consensus::ensure_compatible_tx_signer(ctx)?;
```

Comment:
Would be nice to have a unit test that covers this case.

Reply:
yeah. let me check if/how we currently cover this check

Reply:
I already added the tests ;-)

Reply:
oh great, thanks!